# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/windows/WinCompositorWindowThread.h
# Commit: 0b661055a25e
# Full Hash: 0b661055a25ef73ff81c812b2fe671935aebd5f7
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-01-21 09:30:30
# Description:
#   Bug 1809542: Remove stack-allocated param to runnable. r=rkraesig
#   
#   This directly incorporates the synchronization primitives of the
#   SynchronousTask into the WinCompositorWindowThread object. This removes
#   the paramater to ShutdownTask and allows the wait to timeout correctly.
# ==============================================================================

diff -r d9154943e83f -r 0b661055a25e widget/windows/WinCompositorWindowThread.h
--- a/widget/windows/WinCompositorWindowThread.h	Fri Jan 20 19:21:40 2023 +0000
+++ b/widget/windows/WinCompositorWindowThread.h	Fri Jan 20 19:30:54 2023 +0000
@@ -9,13 +9,9 @@
 
 #include "base/thread.h"
 #include "base/message_loop.h"
+#include "mozilla/Monitor.h"
 
 namespace mozilla {
-
-namespace layers {
-class SynchronousTask;
-}
-
 namespace widget {
 
 struct WinCompositorWnds {
@@ -53,11 +49,12 @@
 
  private:
   explicit WinCompositorWindowThread(base::Thread* aThread);
-  ~WinCompositorWindowThread();
+  ~WinCompositorWindowThread() {}
 
-  void ShutDownTask(layers::SynchronousTask* aTask);
+  void ShutDownTask();
 
-  base::Thread* const mThread;
+  UniquePtr<base::Thread> const mThread;
+  Monitor mMonitor;
 };
 
 }  // namespace widget
